mdit_py_plugins/attrs/index.py

```diff
@@ -58,6 +65,12 @@ def _attr_inline_rule(state: StateInline, silent: bool) -> bool:
             return False
         try:
             new_pos, attrs = parse(state.src[state.pos :])
+            if allowed_attributes:
+                attrs = {k: v for k, v in attrs.items() if k not in allowed_attributes}
+                if attrs:
+                    raise NotAllowedAttributesError(
```

Comment:
err, I'm a little hesitant to add in intentional exceptions:
Markdown is generally designed to be fault-tolerant, i.e. parsing never fails. This is e.g. the case for HTML tags; unsecure ones are just ignored, rather than excepted.

I would simply remove the unsecure attributes from the `attrs`, then store them on e.g. `attr_token.meta["insecure"_attrs"]`.
Then external programs could access them, if they desire to warn about them,
but otherwise parsing will succeed

Reply:
fixed